src/transaction/TransactionResponse.js

```diff
@@ -170,10 +214,34 @@ export default class TransactionResponse {
         };
     }
 
+    /**
+     *
+     * @param {Client} client
+     * @returns {Promise<TransactionReceipt>}
+     */
+    _retryTransaction(client) {
+        if (!client.operatorAccountId) {
+            throw new Error("Operator account is not set");
+        }
+
+        this.transactionId = TransactionId.generate(client.operatorAccountId);
```

Comment:
what if the transaction fails. but the transaction id was not generated from the client?